architecture/transaction-layer.md

```diff
@@ -0,0 +1,204 @@
+
+However, when timestamps are moved, it introduces anomalies into the database. This is an intuitive concept, that it violates the notion of strict serialiazability to reorder the state of your data's values.
+
+In this document, you can find more about our [isolation levels]().
+
+### Timestamp Cache
+
+To provide provide serializability, whenever an operation reads a value, we store the operation's timestamp in a Timestamp Cache, which shows the high-water mark for values being read.
+
+Whenever writes occur, their timestamps are checked against the TimestampCache to ensure its timestamp is greater than the Timestamp Cache (otherwise we would allow you to rewrite history, which produces anomalies). If the timestamp is less than, we attempt to move the timestamp to a time later timestamp than the cache.
+
+### client.Txn and TxnCoordSender
+
+As we mentioned in the SQL layer's architectural overview, CockroachDB converts all SQL statements into key-value (KV) operations, which is how data is ultimately stored and accessed. 
+
+All of the KV operations generated from the SQL layer's use `client.Txn`, which is the transactional interface for the CockroachDB KV layer––but, as we discussed above, all statements are treated as transactions, so all statements use this interface.
+
+However, `client.Txn` is actually just a wrapper around `TxnCoordSender`, which plays a crucial role in our code base by:
+
+- Dealing with transactions' state. After a transaction is started, `TxnCoordSender` starts asynchronously sending heartbeat messages to that transaction's Transaction Record, which signal that it should be kept alive. If the `TxnCoordSender`'s heartbeating stops, the Transaction Record is moved to `ABORTED` status.
+- Tracking each written key or key range over the course of the transaction. 
+- Clearing the accumulated Write Intent for the transaction when it's committed or aborted. All requests being performed as part of a transaction have to go through the same `TxnCoordSender` to account for all of its Write Intents, which optimizes the cleanup process.
+
+After setting up this bookkeeping, the request is passed to the `DistSender` in the Distribution Layer.
+
+### Transaction Records
+
+When a transaction starts, `TxnCoordSender` writes a Transaction Record to the range containing the first key modified in the transaction. As mentioned above, the Transaction Record provides the system with a source of truth about the status of a transaction.
+
+The Transaction Record expresses one of the following dispositions of a transaction:
+
+- `PENDING`: The initial status of all values, indicating that the transaction the Write Intent belongs to is in progress. Note that retried transactions remain in this status because we continue to increment the transaction's priority in an attempt to cause transaction's likelihood of succeeding to increase the longer they remain in the system.
```

Comment:
Let's stick with the singular here:

> Note that a retried transaction remains in this status because we continue to increment its priority in an attempt to cause the transaction's likelihood of succeeding to increase the longer it remains in the system.`